### PR TITLE
Variable Clarity

### PR DESCRIPTION
**Pull request by openbots-ff**
Input and Output variables must all be declared/modified by enclosing them in curly brackets.

Variable Methods have been cleaned up and obsolete code has been removed. All throughout the taskt commands, ConvertUserVariableToString()/ConvertUserVariableToObject() are now used to read variables, while StoreInUserVariable() is used to write to all types of variables. 

Additional cleanup has been done for all output variables in all commands as well as for app instances. 

Commands other than RunTaskCommand should no longer be allowed to reference/modify the engine's variable list directly. They must instead use one of the methods mentioned above.

### CODE REVIEW

**Review by CameronHerwig — DISMISSED** [review]

**Review by CameronHerwig — DISMISSED** [review]

File: taskt.Studio/Commands/Misc Commands/ShowMessageCommand.cs [hunk 1]
```diff
@@ -16,39 +15,39 @@ namespace taskt.Commands
...
         [XmlAttribute]
-        [PropertyDescription("Close After X (Seconds) - 0 to bypass")]
-        [InputSpecification("Specify how many seconds to display on screen. After the amount of seconds passes, the message box will be automatically closed and script will resume execution.")]
-        [SampleUsage("**0** to remain open indefinitely or **5** to stay open for 5 seconds.")]
-        [Remarks("")]
-        public int v_AutoCloseAfter { get; set; }
+        [PropertyDescription("Close After X (Seconds)")]
+        [InputSpecification("Specify how many seconds to display the message on screen. After the specified time," + 
+                            "\nthe message box will be automatically closed and script will resume execution.")]
+        [SampleUsage("5 || {vSeconds} || 0)")]
```
Comment (ob-ab) [inline taskt.Studio/Commands/Misc Commands/ShowMessageCommand.cs]:
```suggestion
        [SampleUsage("0 || 5 || {vSeconds})")]
```

File: taskt.Core/Script/ScriptVariable.cs [hunk 2]
```diff
@@ -33,73 +27,5 @@ public class ScriptVariable
         /// <summary>
         /// retrieve value of the variable
         /// </summary>
```
Comment (ob-ab) [inline taskt.Core/Script/ScriptVariable.cs]: Please remove these comments.

File: taskt.Commands/Excel Commands/ExcelGetRangeCommand.cs [hunk 3]
```diff
@@ -26,9 +26,8 @@ public class ExcelGetRangeCommand : ScriptCommand
         [XmlAttribute]
```
Comment (ob-ab) [inline taskt.Commands/Excel Commands/ExcelGetRangeCommand.cs:26]: As you have removed the **Delimited String** (output option), please update the Class `Description` at line 22.

**Review by ob-ab — COMMENTED** [review]
Please make a few minor changes.

File: taskt.Commands/Excel Commands/ExcelGetRangeCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt.Commands/Excel Commands/ExcelGetRangeCommand.cs:26]: done

File: taskt.Core/Script/ScriptVariable.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt.Core/Script/ScriptVariable.cs]: done

**Review by CameronHerwig — APPROVED** [review]

**Review by ob-ab — APPROVED** [review]

**Pull request merged**